src/sidebar/components/MarkdownView.tsx

```diff
@@ -35,21 +76,49 @@ export default function MarkdownView({
     });
   }, [markdown]);
 
+  useEffect(() => {
+    const listenerCollection = new ListenerCollection();
+    const foundMentions = replaceMentionTags(content.current!, mentions);
+
+    for (const [mention, element] of foundMentions) {
+      listenerCollection.add(element, 'mouseenter', () => {
+        setActiveMention(mention);
+        mentionsPopoverAnchorRef.current = element;
+      });
+      listenerCollection.add(element, 'mouseleave', () => {
+        setActiveMention(null);
+        mentionsPopoverAnchorRef.current = null;
+      });
+    }
+
+    return () => listenerCollection.removeAll();
+  }, [html, mentions]);
+
   // NB: The following could be implemented by setting attribute props directly
   // on `StyledText` (which renders a `div` itself), versus introducing a child
   // `div` as is done here. However, in initial testing, this interfered with
   // some overflow calculations in the `Excerpt` element. This could be worth
   // a review in the future.
   return (
     <div className="w-full break-anywhere cursor-text">
-      <StyledText>
+      <StyledText classes="relative">
         <div
           className={classes}
           data-testid="markdown-text"
           ref={content}
           dangerouslySetInnerHTML={{ __html: html }}
           style={style}
         />
+        {hasMentions && (
+          <Popover
+            open={!!activeMention}
+            onClose={() => setActiveMention(null)}
+            anchorElementRef={mentionsPopoverAnchorRef}
+            classes="p-2"
+          >
+            {activeMention?.display_name ?? activeMention?.username}
```

Comment:
I know this is just an initial implementation, but I would render both the username and the display name. For the LMS use case we'll obviously need to use just the display name.